src/System.Collections/tests/Generic/Dictionary/Dictionary.Generic.Tests.netcoreapp.cs

```diff
@@ -6,6 +6,38 @@ namespace System.Collections.Tests
 {
     public abstract partial class Dictionary_Generic_Tests<TKey, TValue> : IDictionary_Generic_Tests<TKey, TValue>
     {
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void Dictionary_Generic_TryAdd(int count)
+        {
+            var dictionary = (Dictionary<TKey, TValue>)GenericIDictionaryFactory(count);
+
+            TKey key = CreateTKey(seed: count);
+            TValue value = dictionary.ContainsKey(key) ? dictionary[key] : default(TValue);
+            int originalCount = dictionary.Count;
+            bool alreadyContainsKey = dictionary.ContainsKey(key);
```

Comment:
You could do:

```cs
bool alreadyContainsKey = dictionary.ContainsKey(key);
TValue value = alreadyContainsKey ? dictionary[key] : default(TValue);
...
Assert.Equal(!alreadyContainsKey, dictionary.TryAdd(key, default(TValue)));
```

I'm with @stephentoub that there is no need in calling `ContainsKey(key)` twice.